x-pack/plugins/apm/scripts/kibana-security/create_kibana_user_role.ts

```diff
@@ -0,0 +1,534 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import axios, { AxiosRequestConfig, AxiosError } from 'axios';
+import { union, difference, once } from 'lodash';
+
+/* eslint-disable no-console */
+
+interface User {
+  username: string;
+  roles: string[];
+  full_name?: string;
+  email?: string;
+  enabled?: boolean;
+}
+
+export async function createKibanaUserRole({
+  kibanaRoleSuffix,
+  esUserName,
+  esPassword,
+  kibanaBaseUrl,
+}: {
+  esUserName: string;
+  kibanaRoleSuffix?: string;
+  esPassword?: string;
+  kibanaBaseUrl?: string;
+}) {
+  if (!esPassword) {
+    console.log(
+      'Please specify credentials for elasticsearch: `--username elastic --password abcd` '
+    );
+    return;
+  }
```

Comment:
These checks should only be for the cli part. `esPassword` and `kibanaBaseUrl` should be required here.